src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -91,7 +91,17 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     is_linux = plan_info.reserved
     node_default_version = NODE_VERSION_DEFAULT
     location = plan_info.location
-    site_config = SiteConfig(app_settings=[])
+    # This is to keep the existing appsettings for a newly created webapp on existing webapp name.
+    name_validation = client.check_name_availability(name, 'Site')
+    if not name_validation.name_available:
+        existing_app_settings = _generic_site_operation(cmd.cli_ctx, resource_group_name,
+                                                        name, 'list_application_settings')
+        settings = []
```

Comment:
why do we need this?  if no app_settings exist does this return [] if not you can simply do a check if existing_app_settings is [] then site_config =SiteConfig(app_settings=[]) else  site_config =SiteConfig(app_settings=existing_app_settings) ?

Reply:
@panchagnula - `existing_app_settings` is of type `dictionary`. We cannot assign it directly to app_settings, which is of type `NamedValuePair`. So we need to convert it into `NamedValuePair`.